### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
I can drop/split/collapse things w/in reason.
Most changes are to comments, some are to local variables, some are to output, a couple are to enums

### CODE REVIEW

File: apps/openssl.cnf [hunk 1]
```diff
@@ -136,7 +136,7 @@ stateOrProvinceName_default	= Some-State
 localityName			= Locality Name (eg, city)
 
 0.organizationName		= Organization Name (eg, company)
-0.organizationName_default	= Internet Widgits Pty Ltd
+0.organizationName_default	= Internet Widgets Pty Ltd
```
Comment (kroeckx) [inline apps/openssl.cnf]: I'm not sure we should change this or not. (Same in openssl-vms.cnf)

File: crypto/engine/eng_list.c [hunk 2]
```diff
@@ -18,7 +18,7 @@
 /*
  * The linked-list of pointers to engine types. engine_list_head incorporates
  * an implicit structural reference but engine_list_tail does not - the
- * latter is a computational niceity and only points to something that is
+ * latter is a computational nicety and only points to something that is
```
Comment (kroeckx) [inline crypto/engine/eng_list.c]: necessity?

File: include/openssl/asn1.h [hunk 3]
```diff
@@ -40,7 +40,7 @@ extern "C" {
 
 # define V_ASN1_CONSTRUCTED              0x20
 # define V_ASN1_PRIMITIVE_TAG            0x1f
-# define V_ASN1_PRIMATIVE_TAG            0x1f
+# define V_ASN1_PRIMITIVE_TAG            0x1f
```
Comment (kroeckx) [inline include/openssl/asn1.h]: Note that the correct one is already above is. This is for compatibility reasons because this is a public header file. Maybe we should add a comment explaining that?

File: include/openssl/cms.h [hunk 4]
```diff
@@ -491,14 +491,14 @@ int ERR_load_CMS_strings(void);
 # define CMS_R_TYPE_NOT_ENVELOPED_DATA                    146
 # define CMS_R_UNABLE_TO_FINALIZE_CONTEXT                 147
 # define CMS_R_UNKNOWN_CIPHER                             148
-# define CMS_R_UNKNOWN_DIGEST_ALGORIHM                    149
+# define CMS_R_UNKNOWN_DIGEST_ALGORITHM                    149
```
Comment (kroeckx) [inline include/openssl/cms.h]: As above, you need to keep the incorrect name too. I'm not sure how our tools deal with this.

File: include/openssl/obj_mac.h [hunk 5]
```diff
@@ -856,9 +856,9 @@
 #define NID_id_smime_aa_contentReference                221
 #define OBJ_id_smime_aa_contentReference                OBJ_id_smime_aa,10L
 
-#define SN_id_smime_aa_encrypKeyPref            "id-smime-aa-encrypKeyPref"
-#define NID_id_smime_aa_encrypKeyPref           222
-#define OBJ_id_smime_aa_encrypKeyPref           OBJ_id_smime_aa,11L
+#define SN_id_smime_aa_encryptKeyPref            "id-smime-aa-encryptKeyPref"
+#define NID_id_smime_aa_encryptKeyPref           222
+#define OBJ_id_smime_aa_encryptKeyPref           OBJ_id_smime_aa,11L
```
Comment (kroeckx) [inline include/openssl/obj_mac.h]: This is also a public header file ...

File: include/openssl/cms.h (hunk 4 above)
Comment (jsoref) [inline include/openssl/cms.h]: My tool is me, a /* */ comment would be enough for me to leave something alone. I can certainly add in the compat item here.

File: include/openssl/obj_mac.h (hunk 5 above)
Comment (jsoref) [inline include/openssl/obj_mac.h]: Worse, it's a public signature. I saw. I'll let you decide what to do about it.

For expediency, I can split those changes out to a different PR if that helps.

**Comment by jsoref** [comment]
Could someone kick the cla bot? This change falls under the cla exception. That said, if it had provided better instructions, I'm not opposed to signing...

**Comment by richsalz** [comment]
You have to put "CLA: trivial" in your commit text.  (The latest commit only.)  And then the two people who review the code have to agree that it's trivial.

And all of these are going to be merged/squashed I expect.

File: crypto/engine/eng_list.c (hunk 2 above)
Comment (jsoref) [inline crypto/engine/eng_list.c]: I believe that the author meant "nicety", but I think "optimization" is probably a clearer word choice.

**Comment by jsoref** [comment]
I don't mind squashing. but I don't want to do it until everything is done. It's much easier to peel things out when they're separate, e.g. the widget fix is a single commit which I've moved to its own branch. If I did this all as a single commit, I'd have had to fish for all of the pieces and try to tease them apart.

I'm also not a huge fan of magic words in various places. For the time being, I'm going to ignore your bot. If I put the commit message in the last commit in my sequence, and that moves (which actually happened here), then I have to redo it. And if I were to take the approach of sprinkling it everywhere, then the squash step gets messier.

**Comment by jsoref** [comment]
ok, everything that @kroeckx commented about has been split off.
Is everyone happy with the remainder?
Would I need to be the one to do the collapse, or can someone else? (If someone else does it, I presume they can also add the `CLA: trivial`.)

**Comment by kroeckx** [comment]
I have no problem with calling some trivial if it's just a few
spelling fixes. But the amount of fixes I see here is above what I
would call trivial, you clearly put a lot of effort in this.

**Comment by jsoref** [comment]
It's actually pretty trivial, a script generates words, and then I look for the instances, and basically let Google Suggest the corrected spelling. It's fairly mindless actually.

But, I can go ahead and sign the CLA. I don't suppose you have a version that doesn't require me to use ink (my printer is running low...).

**Comment by jsoref** [comment]
Do I need to do something forget to appease the bot for cla-check?

**Comment by richsalz** [comment]
now that we have your CLA on file, it's automatic.  the first time, if you don't mark the commit trivial, it has to be manually removed.  so no, nothing for you to do.

File: include/openssl/cms.h [hunk 6]
```diff
@@ -498,7 +498,7 @@ int ERR_load_CMS_strings(void);
 # define CMS_R_UNSUPPORTED_KEK_ALGORITHM                  153
 # define CMS_R_UNSUPPORTED_KEY_ENCRYPTION_ALGORITHM       179
 # define CMS_R_UNSUPPORTED_RECIPIENT_TYPE                 154
-# define CMS_R_UNSUPPORTED_RECPIENTINFO_TYPE              155
+# define CMS_R_UNSUPPORTED_RECIPIENTINFO_TYPE              155
```
Comment (kroeckx) [inline include/openssl/cms.h]: This is also a public header file.

**Comment by richsalz** [comment]
The way to deal with "corrected" spelling errors in public header files is to put the correct spelling BEFORE the "begin error codes" label.  (Or put the bogus spelling before)  And define one to be the other.  I don't have a strong preference, but any changes you make must survive a "make update" :)

**Comment by jsoref** [comment]
I've pushed cms to #3463

**Comment by dot-asm** [comment]
This needs re-base. But I'm also apposed to the idea of grouping commits by misspelled word. Just like with CMS it should rather be by file or group of files that have something in common, than bunch of unrelated files with similar spelling error...

**Comment by jsoref** [comment]
@dot-asm: suggest a split. I can trivially resegment. It's much harder to handle merge conflicts after I do that...

**Comment by kroeckx** [comment]
I suggest we just turn this into 1 big commit once someone else reviewed it.

File: LICENSE [hunk 7]
```diff
@@ -74,7 +74,7 @@
  * The implementation was written so as to conform with Netscapes SSL.
  * 
  * This library is free for commercial and non-commercial use as long as
- * the following conditions are aheared to.  The following conditions
+ * the following conditions are adhered to.  The following conditions
```
Comment (FdaSilvaYY) [inline LICENSE]: For some legal reasons, don't touch any word of license text, please.

File: LICENSE [hunk 8]
```diff
@@ -99,7 +99,7 @@
  *    must display the following acknowledgement:
  *    "This product includes cryptographic software written by
  *     Eric Young (eay@cryptsoft.com)"
- *    The word 'cryptographic' can be left out if the rouines from the library
+ *    The word 'cryptographic' can be left out if the routines from the library
```
Comment (FdaSilvaYY) [inline LICENSE]: Idem !

File: LICENSE (hunk 7 above)
Comment (jsoref) [inline LICENSE]: http://publib.boulder.ibm.com/tividd/td/TWS/SC32-1267-00/en_US/HTML/eqqm1mst68.htm

File: LICENSE (hunk 8 above)
Comment (jsoref) [inline LICENSE]: ibid

**Comment by richsalz** [comment]
@FdaSilvaYY is correct.  Do *not* alter the wording of the SSLeay license.  If nothing else, this is the team's policy.

File: LICENSE (hunk 7 above)
Comment (richsalz) [inline LICENSE]: undo this change.

File: LICENSE (hunk 8 above)
Comment (richsalz) [inline LICENSE]: undo this change.

File: crypto/des/asm/des-586.pl [hunk 9]
```diff
@@ -15,7 +15,7 @@
 require "cbc.pl";
 require "desboth.pl";
 
-# base code is in microsft
+# base code is in microsoft
```
Comment (richsalz) [inline crypto/des/asm/des-586.pl]: Microsoft (I mean, if you're fixing typo's:)

File: engines/e_padlock.c [hunk 10]
```diff
@@ -188,7 +188,7 @@ struct padlock_cipher_data {
             int rounds:4;
             int dgst:1;         /* n/a in C3 */
             int align:1;        /* n/a in C3 */
-            int ciphr:1;        /* n/a in C3 */
+            int cipher:1;       /* n/a in C3 */
```
Comment (richsalz) [inline engines/e_padlock.c]: as with ```dgst``` this is the name assinged by the vendor; undo this change.

File: test/README.external [hunk 11]
```diff
@@ -126,7 +126,7 @@ explicitly run (with more debugging):
 
 $ VERBOSE=1 make TESTS=test_external_krb5 test
 
-Test failures supressions
+Test failures suppressions
 -------------------------
```
Comment (richsalz) [inline test/README.external]: add a dash since at line 130 since you made the line at 129 longer.

**Comment by richsalz** [comment]
This is not trivial.  We will need you to sign our CLA.  Also, if the manpage edits were in a single separate commit, it would be easier to cherry-pick that to previous releases -- the code-internal changes are not required.

File: crypto/des/asm/des-586.pl (hunk 9 above)
Comment (jsoref) [inline crypto/des/asm/des-586.pl]: Happy to do that. fwiw:
```
krb5/src/lib/gssapi/krb5/gssapiP_krb5.h:    SGN_ALG_HMAC_MD5              = 0x0011, /* microsoft w2k;  */
krb5/src/lib/gssapi/krb5/gssapiP_krb5.h:    SEAL_ALG_MICROSOFT_RC4   = 0x0010, /* microsoft w2k;  */
krb5/src/lib/krb5/ccache/cc_mslsa.c: * Implementation of microsoft windows lsa credentials cache
krb5/src/lib/krb5/ccache/cc_mslsa.c: * Closes the microsoft lsa cache, invalidates the id, and frees any resources
krb5/src/util/verto/ev.c:    /* apparently, microsoft broke all the memory barrier stuff in Visual Studio 2008... */
krb5/src/util/verto/ev.c:#define EMPTY       /* required for microsofts broken pseudo-c compiler */
krb5/src/util/verto/ev.c:  /* have to use the microsoft-never-gets-it-right macro */
krb5/src/util/verto/ev_select.c:       * provided (this is documented). what microsoft doesn't tell you that this bug
```

**Comment by jsoref** [comment]
I submitted the CLA a while ago and you accepted it...

I can split out the man pages parts. I'll push an update based on the current comments and then split out the man pages.

**Comment by jsoref** [comment]
I've pushed the doc changes to #3580 
I can drop them from here (or wait for that to land and then let a rebase drop them).

**Comment by richsalz** [comment]
I don’t care about the others.  I find it ironic that a PR about spelling ☺

**Comment by jsoref** [comment]
Hey, my favorites are when commit messages/change logs about misspellings are misspelled.

I actually have a separate commit message for fixing branding. For Microsoft and similar, sometimes it's in a URL where it makes sense to leave things alone.

The one I tend to fix the most often is JavaScript without the capitalized letter, since it's definitely wrong.

After you guys accept a couple of these, I can see about a branding PR – if you're open to it...

File: crypto/des/asm/des-586.pl (hunk 9 above)
Comment (kaduk) [inline crypto/des/asm/des-586.pl]: Those are not in openssl-maintained code; those files are pulled in from krb5/krb5 for their external test suite.  (You could probably make a pull request for them there, though.)

**Comment by richsalz** [comment]
@jsoref can you rebase against a recent master and push?
```
git rebase master
/* fix conflicts */
git push -f github HEAD
```
thanks!

**Comment by jsoref** [comment]
I use hg. I'll rebase, but right now I don't have a computer. This will have to wait, possibly a couple of weeks. I'm sorry.

**Comment by FdaSilvaYY** [comment]
@jsoref: are you able to rebase this PR ?

File: test/README.external [hunk 12]
```diff
@@ -127,7 +127,7 @@ explicitly run (with more debugging):
 $ VERBOSE=1 make TESTS=test_external_krb5 test
 
 Test-failures suppressions
--------------------------
+--------------------------
```
Comment (jsoref) [inline test/README.external]: This isn't a spelling fix...

**Comment by jsoref** [comment]
@richsalz: sorry for the delay; @FdaSilvaYY: done...

**Comment by FdaSilvaYY** [comment]
Ping @openssl

**Comment by jsoref** [comment]
@richsalz ?

**Comment by richsalz** [comment]
Whew, merged.  Squashed to one commit.  Do you have an automated script for this that we could use?
Thank you!

**Comment by jsoref** [comment]
https://github.com/jsoref/spelling
`f` generates the list of potentially misspelled words.
You could try `fchurn`, because for an existing repository, you don't really care about existing terms, just newly introduced terms since the previous time `f` ran (which is 4e74af1b59e5c175d034ec78226e9e14834d3ba2).
